lib/reverse_sentence.rb

```diff
@@ -1,6 +1,45 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear, as string is reversed in place
+# Space complexity: constant, as variables are updated without need for extra space
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return my_sentence if my_sentence == nil || my_sentence.length == 0
+
+  temp = 0
+  left = 0
+  right = my_sentence.length - 1
+
+  while left < right
```

Comment:
Style nit: this looks like it's reversing the entire sentence (which is itself a string). Moving this into a helper method might make things less repetitive + easier to read. 🙂